src/main/scala/main.scala

```diff
@@ -54,4 +52,27 @@ object RockSlicer {
     jsonBlocks.saveAsTextFile("blocks.json")
     sc.stop()
   }
+
+  // Produce joints that achieve some lead balancing when generating the initial Block RDD
+  private def generateSeedJoints(joints: Seq[Joint], numSeedJoints: Integer): (Seq[Joint], Seq[Joint]) = {
+    val (persistentJoints, nonPersistentJoints) = joints.partition(_.shape.isEmpty)
+    persistentJoints.length match {
+      // All of the seed joints will be persistent
+      case len if len >= numSeedJoints => (persistentJoints.take(numSeedJoints),
+          persistentJoints.drop(numSeedJoints) ++ nonPersistentJoints)
+      // All persistent joints are used as seed joints, along with some non-persistent joints
+      case len =>
+          // Sort non-persistent joints by relative position along x axis
+          val sortedNonPersistent = nonPersistentJoints.sortWith(_.centerX < _.centerX)
+          val numNonPersistent = numSeedJoints - len
+          val step = sortedNonPersistent.length / numNonPersistent
+          val seedNonPersistentCandidates = (sortedNonPersistent.indices by step) collect sortedNonPersistent
```

Comment:
Okay, that makes sense. Based that though, it still looks like you're not grabbing a joint that is located close to the center of the domain. In your example, you're grabbing the first element in the `lst` which would correspond to the smallest value of `centerX`. To try and keep things balanced, should it be grabbing something close to the center of the range?